mock_service.go

```diff
@@ -31,33 +32,58 @@ type MockEndpoint struct {
 	// RequestBody     string            `json:"requestBody"`
 }
 
-func New(createMockEndpoint string) *MockService {
+type MockEndpointSeries []MockEndpoint
+
+func New(mockCreatorEndpoint string) *MockService {
 	return &MockService{
-		createMockEndpoint: createMockEndpoint,
-		mockEndpoints:      map[string]map[string]MockEndpoint{},
+		mockCreatorEndpoint:  mockCreatorEndpoint,
+		mockedEndpoints:      map[string]map[string]MockEndpoint{},
 	}
 }
 
 func (m *MockService) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	if req.Method == http.MethodPost && req.URL.Path == m.createMockEndpoint {
+	if req.Method == http.MethodPost && req.URL.Path == m.mockCreatorEndpoint {
 		m.serveRegistrationHTTP(w, req)
 		return
 	}
 
 	m.serveMockHTTP(w, req)
 }
 
+func (m *MockService) PreloadEndpointsFromConf(confFilePath string) (error, bool){
+    if  len(m.mockCreatorEndpoint) < 1 {
+        return  ErrCreatorNotExist, false
+    }
+    fi, err := os.Open(confFilePath)
+    if err != nil{panic(err)}
+    defer fi.Close()
+    fd, _ := ioutil.ReadAll(fi)
+    var series MockEndpointSeries
+    if err := json.Unmarshal(fd, &series); err != nil {
+		log.Fatal("Unable to Unmarshal request body %s: %s", fd, err)
+        return err, false
+    }
+    for _, endpoint := range(series) {
+        m.CreateMockEndpoint(endpoint)
+    }
+    return nil, true
+}
+
 func (m *MockService) CreateMockEndpoint(endpoint MockEndpoint) {
+    if "POST" == endpoint.Method && m.mockCreatorEndpoint == endpoint.Endpoint {
```

Comment:
I can see why this is added to protect this exported method from being called by a consumer who may not know that  registering on top of the `mockRegistrationEndpoint` will not be honored. I don't have a good way to solve this just yet. Perhaps we can un-export to protect it from those callers.